tjhsst/apps/labs/views.py

```diff
@@ -144,6 +149,43 @@ def new(request):
         else:
             form = LabCreationForm()
 
-        return render(request, "labs/new.html", {"lab_form": form})
+        return render(request, "labs/new.html", {"form": form})
+    else:
+        raise Http404
+
+
+def show_projects(request, lab_url):
+    lab = get_object_or_404(Lab, url=lab_url)
+    projects = lab.senior_projects_sponsored.all()
+
+    return render(request, "labs/show_projects.html", {"projects": projects, "lab": lab})
+
+
+def add_project(request):
+    if request.user.is_authenticated:
+        if request.method == "POST":
+            form = ProjectCreationForm(request.POST, request.FILES)
+            if form.is_valid():
+                form.save()
+                return redirect("labs:index")
+        else:
+            form = ProjectCreationForm()
+
+        return render(request, "labs/new.html", {"form": form})
+    else:
+        raise Http404
```

Comment:
If we want (do we?), we could use `CreateView` along with `UserPassesTestMixin`.

Or, I believe that we should add `@login_required` instead of using the `if`.